orchagent/portsorch.cpp

```diff
@@ -394,12 +394,23 @@ bool PortsOrch::allPortsReady()
     return m_initDone && m_pendingPortSet.empty();
 }
 
-/* Upon receiving PortInitDone, all the configured ports have been created*/
+/* Upon receiving PortInitDone, all the configured ports have been created in both hardware and kernel*/
 bool PortsOrch::isInitDone()
 {
     return m_initDone;
 }
 
+// Upon m_portConfigState transiting to PORT_CONFIG_DONE state, all physical ports have been "created" in hardware.
+// Because of the asynchronous nature of sairedis calls, "create" in the strict sense means that the SAI create_port()
+// function is called and the create port event has been pushed to the sairedis pipeline. Because sairedis pipeline
+// preserves the order of the events received, any event that depends on the physical port being created first, e.g.,
+// buffer profile apply, will be popped in the FIFO fashion, processed in the right order after the physical port is
+// physically created in the ASIC, and thus can be issued safely when this function call returns true.
+bool PortsOrch::isConfigDone()
```

Comment:
We have priorities in our Select in swss-common. So it might be not FIFO

Reply:
are you two referring two the same fifo? I think wenda is referring to SAI redis command queue is FIFO. you are referring to the different orch scheduling queue, which is not FIFO.

Reply:
Both of them using the same sonic-swss-common library, which has priorities inside of it.
Check for `swss::Select` inside of sairedis.

Reply:
no, but for the redis command queue, there is only one queue.

Reply:
there maybe some misunderstanding here. let's talk offline.